flight/tasks/encodings.py

```diff
@@ -0,0 +1,103 @@
+"""A class used for encoding/decoding lookups."""
```

Comment:
I will just say that I do not find this to be a very elegant solution. At this point there is no time to change it, but it is not very readable aside from the comments, and code should not only be readable by the inlayed comments. Also, the bidirectional dictionary is in my mind also not an elegant solution to the problem. I still believe the classed system to be the best approach, if only needing a little bit more work. Either way, this is the move for now, I suppose. Don't try to reinvent the wheel with new data structures if you don't have to.